pyani_plus/db_orm.py

```diff
@@ -410,7 +411,26 @@ def cache_comparisons(self) -> None:
         cov_query = np.full([size, size], np.nan, float)
         aln_length = np.full([size, size], np.nan, float)
         sim_errors = np.full([size, size], np.nan, float)
-        for comp in self.comparisons():
+
+        # Using .comparisons() loads all the fields taking much more memory:
+        run_query = aliased(RunGenomeAssociation, name="run_query")
+        run_subjt = aliased(RunGenomeAssociation, name="run_subject")
+        db_query = (
+            select(
+                Comparison.query_hash,
+                Comparison.subject_hash,
+                Comparison.identity,
+                Comparison.cov_query,
+                Comparison.aln_length,
+                Comparison.sim_errors,
+            )
+            .where(Comparison.configuration_id == self.configuration_id)
+            .join(run_query, Comparison.query_hash == run_query.genome_hash)
+            .join(run_subjt, Comparison.subject_hash == run_subjt.genome_hash)
+            .where(run_query.run_id == self.run_id)
+            .where(run_subjt.run_id == self.run_id)
+        )
+        for comp in object_session(self).execute(db_query):
```

Comment:
Looking at this and imagining future even larger use-cases, we might try using a loop over the subject hashes, and make separate queries for each of them. Each query would be simpler (one less join) as well as smaller, and we could avoid one dictionary lookup per entry to find the subject row number too.